test/functional/treesitter/language_spec.lua

```diff
@@ -67,5 +68,15 @@ describe('treesitter API', function()
     end
     eq({true,true}, {has_named,has_anonymous})
   end)
+
+  it('checks if vim.treesitter.get_parser tries to create a new parser on filetype change', function ()
+    command("set filetype=c")
+    -- Should not throw an error when filetype is c
+    eq('c', exec_lua("return vim.treesitter.get_parser(0)._lang"))
```

Comment:
```suggestion
    eq('c', exec_lua("return vim.treesitter.get_parser(0):lang()"))
```
Same thing here then, I assume?

Reply:
done